src/liblzma/meson.build

```diff
@@ -0,0 +1,358 @@
+        'check/crc32_small.c'
+    ]
+else
+    lzma_src += [
+        'check/crc32_table.c'
+    ]
+
+    if have_assembler_x86 == true
+        lzma_src += [
+            'check/crc32_x86.S'
+        ]
+    else
+        lzma_src += [
+            'check/crc32_fast.c'
+        ]
+    endif
+endif
+
+if get_option('checks').contains('crc64') == true
+    if get_option('small') == true
+        lzma_src += [
+            'check/crc64_small.c'
+        ]
+    else
+        lzma_src += [
+            'check/crc64_table.c'
+        ]
+
+        if have_assembler_x86 == true
+            lzma_src += [
+                'check/crc64_x86.S'
+            ]
+        else
+            lzma_src += [
+                'check/crc64_fast.c'
+            ]
+        endif
+    endif
+endif
+
+if get_option('checks').contains('sha256') == true and get_option('external-sha256') == false
+    lzma_src += [
+        'check/sha256.c'
+    ]
+endif
+
+if get_option('encoders').contains('lzma1') == true or
+   get_option('decoders').contains('lzma1') == true or
+   get_option('encoders').contains('lzma2') == true or
+   get_option('decoders').contains('lzma2') == true
+
+    # lz/Makefile.inc
+    if get_option('encoders').contains('lzma1') == true or
+       get_option('encoders').contains('lzma2') == true
+        lzma_src += [
+            'lz/lz_encoder.c',
+            'lz/lz_encoder_mf.c'
+        ]
+    endif
+
+    if get_option('decoders').contains('lzma1') == true or
+       get_option('decoders').contains('lzma2') == true
+        lzma_src += [
+            'lz/lz_decoder.c'
+        ]
+    endif
+endif
+
+if get_option('encoders').contains('lzma1') == true or
+   get_option('decoders').contains('lzma1') == true
+
+    # lzma/Makefile.inc
+    lzma_src += [
+        'lzma/lzma_encoder_presets.c'
+    ]
+
+    if get_option('encoders').contains('lzma1') == true
+        lzma_src += [
+            'lzma/lzma_encoder.c',
+            'lzma/lzma_encoder_optimum_fast.c',
+            'lzma/lzma_encoder_optimum_normal.c'
+        ]
+
+        if get_option('small') == false
+            lzma_src += [
+                'lzma/fastpos_table.c'
+            ]
+        endif
+    endif
+
+    if get_option('decoders').contains('lzma1') == true
+        lzma_src += [
+            'lzma/lzma_decoder.c'
+        ]
+    endif
+
+    if get_option('encoders').contains('lzma2') == true
+        lzma_src += [
+            'lzma/lzma2_encoder.c'
+        ]
+    endif
+
+    if get_option('decoders').contains('lzma2') == true
+        lzma_src += [
+            'lzma/lzma2_decoder.c'
+        ]
+    endif
+
+    # rangecoder/Makefile.inc
+    if get_option('encoders').contains('lzma1') == true
+        lzma_src += [
+            'rangecoder/price_table.c'
+        ]
+    endif
+endif
+
+if get_option('encoders').contains('delta') == true or
+   get_option('decoders').contains('delta') == true
+
+    # delta/Makefile.inc
+    lzma_src += [
+        'delta/delta_common.c'
+    ]
```

Comment:
Same, only this line needs to be inside this if, the rest of the lines can be outside since their comparisons are a subset of this if.